chapters/docstring/docstring.rst

```diff
@@ -0,0 +1,195 @@
+Docstring guide
+==================================
+
+    “Code is more often read than written.”
+
+    — Guido van Rossum
+
+In Jina, we are aware that documentation is an important part of Sofware, but we also think it is especially important for OpenSource. And for this reason, we try extra hard to have clear and extensive documentation for all of our source code. But, at the same time, we know this also takes time and effort, so we want to make things as easy as possible with this guide for you. Here we want to show you the best practices for writing docstrings in Jina.
+
+
+What are docstrings?
+----------------------------------------------------
+
+First, we should define what are we talking about. A docstring is a string literal that we use to document elements of our code, such as functions, methods, modules, and classes. We do this to have a clear understanding about what are the details of each part of our code. We can see it more in detail in `PEP 257 <https://www.python.org/dev/peps/pep-0257/>`_. Jina recommends the following:
+
+* Write docstrings for **public** *functions* and *classes*
+* Optionally you can write docstrings for **private** *functions* and *classes*, but it's not mandatory
+
+In Jina, we use **ReStructuredText** (reST), which is the default markup language used by `Sphinx <https://www.sphinx-doc.org/>`_. You can use *Markdown* too but we encourage you to use reST since *Markdown* doesn't contain rich markup.
+
+
+One-line Docstrings
+----------------------------------------------------
+
+Use one-line docstrings when the description of the class/module/function fits in one line
+
+We suggest the following guidelines:
+
+* Define the Dosctrings with triple-double quotes (""")
```

Comment:
are these a specific style? google / sphinx / numpy?

Reply:
What exaclty?

Reply:
The list of requirements and the style guide in general belong to `sphinx` style guide? It would be easier if we knew, so that we could configure linters later on

Reply:
So the style name is `pep257` Add a note on this and how you can lint locally with http://www.pydocstyle.org/en/stable/

`pydocstyle --convention=pep257 $YOUR_CODE`